src/javascript/app_2/pages/trading/components/elements/language_switcher.jsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { localize } from '../../../../../_common/localize';
+
+class LanguageSwitcher extends React.PureComponent {
+    constructor(props) {
+        super(props);
+        this.state = {
+            is_collapsed   : false,
+            active_language: this.props.languages[0],
+        };
+    }
+
+    toggleLanguageList = () => {
+        this.setState({
+            is_collapsed: !this.state.is_collapsed,
+        });
+    }
+
+    switchLanguage = (lang) => {
+        this.setState({
+            active_language: lang,
+        });
+    }
+
+    render() {
+        const language_list_collapsed = {
+            visibility: `${this.state.is_collapsed ? 'visible' : 'hidden'}`,
+            position  : 'absolute',
+            width     : '100%',
+        };
+
+        const language_is_active = (lang) => {
+            if (lang.id === this.state.active_language.id) {
+                return true;
+            }
+            return false;
+        };
+
+        return (
+            <React.Fragment>
+                <div className='drawer-item' onClick={this.toggleLanguageList}>
+                    <span className={`parent-item ${this.state.active_language.id||''}`}>
+                        {localize('Language')} : {this.state.active_language.name}
+                    </span>
+                </div>
+                <div
+                    className={`lang-switcher-list ${this.state.is_collapsed ? 'show' : ''}`}
+                    style={language_list_collapsed}
+                >
+                    <div className='lang-switcher-list-header' onClick={this.toggleLanguageList}>
+                        <span className='lang-switcher-list-desc'>
+                            {localize('Choose your language')}
+                        </span>
+                    </div>
+                    {this.props.languages.map((language, idx) => (
+                        <React.Fragment key={idx}>
+                            <div
+                                className={`lang-switcher-language ${language_is_active(language) ? 'active' : ''}`}
+                                onClick={this.switchLanguage.bind(null, language)}
```

Comment:
nitpick: `onClick{this.switchLanguage}` should work, since it's a member lambda. (it's not on prototype).

Reply:
`onClick{this.switchLanguage}` is a typo?

Reply:
@zakibinary  yea, there is a missing `=` 😅